internal/server/audit/events.go

```diff
@@ -0,0 +1,165 @@
+package audit
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"go.flipt.io/flipt/rpc/flipt"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	eventVersion          = "0.1"
+	eventVersionKey       = "flipt.event.version"
+	eventActionKey        = "flipt.event.action"
+	eventTypeKey          = "flipt.event.type"
+	eventMetadataActorKey = "flipt.event.metadata.actor"
+	eventPayloadKey       = "flipt.event.payload"
+	eventTimestampKey     = "flipt.event.timestamp"
+)
+
+// Event holds information that represents an action that has occurred in the system.
+type Event struct {
+	Version string `json:"version"`
+
+	Type flipt.Subject `json:"type"`
+
+	Action flipt.Action `json:"action"`
+
+	Metadata Metadata `json:"metadata"`
+
+	Payload interface{} `json:"payload"`
+
+	Timestamp string `json:"timestamp"`
+}
+
+// NewEvent is the constructor for an event.
+func NewEvent(r flipt.Request, actor *Actor, payload interface{}) *Event {
+	return &Event{
+		Version: eventVersion,
+		Action:  r.Action,
```

Comment:
i think i need to make this the past tense of the action to conform to our existing audit events language. ie: `create` -> `created`